app/assets/v2/js/cart.js

```diff
@@ -928,133 +928,207 @@ Vue.component('grants-cart', {
-        saveSubscriptionPayload.sub_new_approve_tx_id.push(donation.tokenApprovalTxHash);
-        saveSubscriptionPayload.token_address.push(tokenAddress);
-        saveSubscriptionPayload.token_symbol.push(tokenName);
-      } // end for each donation
-
-      // to allow , within comments
-      saveSubscriptionPayload.comment = saveSubscriptionPayload.comment.join('_,_');
-
-      // Configure request parameters
-      const url = '/grants/bulk-fund';
-      const headers = {
-        'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'
-      };
-      const saveSubscriptionParams = {
-        method: 'POST',
-        headers,
-        body: new URLSearchParams(saveSubscriptionPayload)
-      };
+        // if (json.failures.length > 0) {
+        //   // Something went wrong, so we create a backup of the users cart
+        //   await this.manageEthereumCartJSONStore(`${userAddress} - ${new Date().getTime()}`, 'save');
+        // }
+
+        // // Clear JSON Store
+        // await this.manageEthereumCartJSONStore(userAddress, 'delete');
+      } catch (err) {
+        // Something went wrong, so we use the manual ingestion process instead
+        console.error(err);
+        console.log('Standard contribution ingestion failed, falling back to manual ingestion');
+        await this.postToDatabaseManualIngestion(txHash, userAddress);
+      }
+    },
+
+    // Alternative to postToDatabase that uses the manual ingestion process
+    async postToDatabaseManualIngestion(txHash, userAddress) {
+      // Determine if this was a zkSync checkout or standard L1 checkout. For this endpoint, we pass a txHash
+      // to ingest L1 contributions or an address to pass L2 contributions
+      const checkout_type = txHash[0].startsWith('sync') ? 'eth_zksync' : 'eth_std';
+      
+      txHash = checkout_type === 'eth_std' ? txHash[0] : ''; // txHash is always an array of hashes from checkout
+      userAddress = checkout_type === 'eth_zksync' ? userAddress : '';
+
+      // Get user's signature to prevent ingesting arbitrary transactions under your own username, then ingest
+      const { signature, message } = await this.signMessage(userAddress);
+      const csrfmiddlewaretoken = document.querySelector('[name=csrfmiddlewaretoken]').value;
+      const url = '/grants/ingest';
+      const headers = { 'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8' };
+      const network = document.web3network || 'mainnet';
+      const payload = { csrfmiddlewaretoken, txHash, userAddress, signature, message, network };
+      const postParams = { method: 'POST', headers, body: new URLSearchParams(payload) };
+      let json; // response
 
       // Send saveSubscription request
-      const res = await fetch(url, saveSubscriptionParams);
-      const json = await res.json();
+      try {
+        _alert('Please be patient, as manual ingestion may take 1-2 minutes', 'info', '3000');
```

Comment:
do we want to tell the user we are even doing manual ingestion?  seems more fluid to me to just abstract it from the suer

Reply:
Up to you. I think we need to indicate it somehow, even if just say "Please be patient, this may take 1-2 minutes". But due to the extra signature required you can't really abstract it away entirely